[git-webkit revert] Setting issue as a radar duplicates the radar in commit message

briannafan opened the pull request:
#### fbca8fd9a3e2b3186d3563de390010a8fce5f840
<pre>
[git-webkit revert] Setting issue as a radar duplicates the radar in commit message
<a href="https://bugs.webkit.org/show_bug.cgi?id=268173">https://bugs.webkit.org/show_bug.cgi?id=268173</a>
<a href="https://rdar.apple.com/problem/121674742">rdar://problem/121674742</a>

Reviewed by Jonathan Bedard.

Changes how radars and automatic commenting are handled by git-webkit revert.

* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/revert.py:
(Revert.get_issue_info): Moved automatic bug comments to Revert.revert_commit.
(Revert.create_revert_commit_msg): Added search for bugs referenced by a radar.
If an issue is redacted, prompt for a reason.
(Revert.revert_commit): Automatic bug comments are added only after a successful revert.
(Revert.main):

Canonical link: <a href="https://commits.webkit.org/273600@main">https://commits.webkit.org/273600@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/23caead1069061c6e5cdd31a9e484d70c1b85518

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/35910 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/14854 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/38103 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/38631 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/32309 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/37128 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/17272 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/11879 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/31064 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/36462 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/12566 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/31906 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/11034 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/36094 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/11044 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/32083 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/39880 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/32626 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/32411 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/36992 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/11214 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/9120 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/35069 "Passed tests") | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/35886 "Passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/12962 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/8193 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/11727 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/12044 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

JonWBedard commented on Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/revert.py:
```diff
@@ -167,30 +163,44 @@ def create_revert_commit_msg(cls, args, commit_objects, **kwargs):
-            if revert_issue_radar:
-                break
+
+        if revert_issue.redacted or isinstance(revert_issue.tracker, radar.Tracker):
```
I think we should put a "FIXME" here. This is the correct behavior for some repositories, but that's really only true if `radar` isn't the primary tracker for the repository, right?

briannafan committed:
[git-webkit revert] Setting issue as a radar duplicates the radar in commit message
https://bugs.webkit.org/show_bug.cgi?id=268173
rdar://problem/121674742

Reviewed by Jonathan Bedard.

Changes how radars and automatic commenting are handled by git-webkit revert.

* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/revert.py:
(Revert.get_issue_info): Moved automatic bug comments to Revert.revert_commit.
(Revert.create_revert_commit_msg): Added search for bugs referenced by a radar.
If an issue is redacted, prompt for a reason.
(Revert.revert_commit): Automatic bug comments are added only after a successful revert.
(Revert.main):

Canonical link: https://commits.webkit.org/273600@main
